frontend/app/module/input/PollingStation/PollingStationLayout.tsx

```diff
@@ -1,82 +1,75 @@
-import { useEffect, useState } from "react";
+import { useState } from "react";
 import { Outlet, useParams } from "react-router-dom";
 
 import { PollingStationProgress } from "app/component/pollingstation/PollingStationProgress";
 
-import {
-  PollingStation,
-  PollingStationFormController,
-  useElection,
-  usePollingStationList,
-} from "@kiesraad/api";
+import { PollingStationFormController, useElection, usePollingStation } from "@kiesraad/api";
 import { IconCross } from "@kiesraad/icon";
 import { Badge, Button, Modal, PollingStationNumber, WorkStationNumber } from "@kiesraad/ui";
 
 export function PollingStationLayout() {
-  const { pollingStationId } = useParams();
-  const { pollingStations } = usePollingStationList();
-  const [pollingStation, setPollingStation] = useState<PollingStation>();
   const { election } = useElection();
+  const { pollingStationId } = useParams();
+  const { pollingStation, loading } = usePollingStation(pollingStationId);
   const [openModal, setOpenModal] = useState(false);
 
   function changeDialog() {
     setOpenModal(!openModal);
   }
 
-  useEffect(() => {
-    const parsedStationId = parseInt(pollingStationId || "0");
-    setPollingStation(
-      pollingStations.find((pollingStation) => pollingStation.id === parsedStationId),
-    );
-  }, [pollingStationId, pollingStations]);
+  if (loading) {
+    return null;
+  }
+
+  if (!pollingStation) {
+    throw Error("Polling station not found");
+  }
 
   return (
-    pollingStation && (
-      <PollingStationFormController
-        election={election}
-        pollingStationId={parseInt(pollingStationId || "0")}
-        entryNumber={1}
-      >
-        <header>
-          <section>
-            <PollingStationNumber>{pollingStation.number}</PollingStationNumber>
-            <h1>{pollingStation.name}</h1>
-            <Badge type="first_entry" />
-          </section>
-          <section>
-            <Button variant="secondary" size="sm" onClick={changeDialog} rightIcon={<IconCross />}>
-              Invoer afbreken
-            </Button>
-            <WorkStationNumber>16</WorkStationNumber>
-          </section>
-        </header>
-        <main>
+    <PollingStationFormController
+      election={election}
+      pollingStationId={parseInt(pollingStationId || "0")}
```

Comment:
```suggestion
      pollingStationId={pollingStation.id}
```

Reply:
Yes, good one, we parse this unnecessarily